src/opencl/error.rs

```diff
@@ -1,30 +1,33 @@
+use opencl3::{device::DeviceInfo, error_codes::ClError, program::ProgramInfo};
+
 #[derive(thiserror::Error, Debug)]
 #[allow(clippy::upper_case_acronyms)]
 pub enum GPUError {
-    #[error("Ocl Error: {0}")]
-    Ocl(ocl::Error),
+    #[error("Opencl3 Error: {0}{}", match .1 {
+       Some(message) => format!(" {}", message),
+       None => "".to_string(),
+    })]
+    Opencl3(ClError, Option<String>),
```

Comment:
why not just do sth like `{.1:?}` and print the debug version?

Reply:
Because this is nicer output. So it won't be `Error: `Some("the whole build log over several lines, i don't even know if it would then do line breaks properly")`.